### PR TITLE
Added page for editing role permissions

### PR DESCRIPTION
**Pull request by Benson272**
### Administrative Info
Monday Board ID: 1181559666
Make sure your branch name conforms to: `<feature/staging/hotfix/...>/[username]/[Monday Item ID]-[3-4 word description separated by dashes]`. Otherwise, please rename your branch and create a new PR.

### Changes
What changes did you make?
Added a page to edit roles and create roles
Connected pages to corresponding backend route

### Testing
How did you confirm your changes worked? 
1) Went to the page to display all roles
2) clicked on the edit button for a role and checked that permissions are correctly displayed
3) edit permissions and clicked save
4) Went back to page that displays all roles and clicked the "+ create role" button
5) Create role and gave random permissions (also tested repeat role name and empty role name)
6) Deleted role I just created by editing the role and clicking the delete button

### Confirmation of Change 
Upload a screenshot, if possible. Otherwise, please provide instructions on how to see the change.

Default permissions for developers
image

Permissions after saving changes
image

Creating a role named Test (code will auto-capitalize role name)
image

Role appears after being created
image

Deleting the Test role
image

Test role gone after being deleted
image

### CODE REVIEW

**Review by AnsonLee99 — CHANGES_REQUESTED** [review]
There is one more button handle for the "Create new role" button where EditRoles should be hooked up.
https://github.com/TritonSE/TSE-Oktavian/blob/d3c75bcdb28652aa61f1c6228f6df4b26559f457/frontend/src/pages/admin/roles/Roles.js#L77

**Comment by AnsonLee99** [comment]
Also, would we want to wait on this PR until Justin finishes his edit role backend routes?

File: frontend/src/components/RolePermissionsCard.js [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+        <Button
+          className={classes.saveButton}
+          variant="contained"
+          color="primary"
+          onClick={(event) => {
+            event.preventDefault();
+            // TODO - Implement save function
+          }}
+        >
+          Save Changes
```
Comment (hahsu) [inline frontend/src/components/RolePermissionsCard.js:222]: This button needs to say Save Changes or Create depending on if we came from editing or creating

**Review by hahsu — CHANGES_REQUESTED** [review]

File: frontend/src/components/RolePermissionsCard.js [hunk 2]
```diff
@@ -0,0 +1,389 @@
...
+  };
+
+  const handleChange = (event) => {
+    const { permissions } = state.role;
+    const field_name = event.target.name;
+    permissions[field_name] = event.target.checked;
+    // unchecks admin box if any indented checkboxes were unchecked
+    if (permissions[field_name] !== null) {
+      permissions["admin"] = false;
+    }
```
Comment (hahsu) [inline frontend/src/components/RolePermissionsCard.js:150]: I'm a bit confused by this, why are we checking for `!=== null?` I assumed that event.target.checked would always be true or false. Also how do you know that permissions[field_name] is an indented field? Don't we only want to only make `permissions["admin"] = false ` if an indented box was unchecked?

**Review by hahsu — CHANGES_REQUESTED** [review]

**Review by AnsonLee99 — APPROVED** [review]
lgtm!

**Review by hahsu — APPROVED** [review]
lgtm, thanks!

**Review by justinyaodu — APPROVED** [review]
Suggested one tiny change, LGTM!
